compiler/src/dotty/tools/dotc/core/Symbols.scala

```diff
@@ -622,17 +634,30 @@ object Symbols {
       denot.asInstanceOf[ClassDenotation]
 
     override protected def prefixString = "ClassSymbol"
+
+    override def enclClassChain(implicit ctx: Context): List[Symbol] =
+      if (this.is(Flags.PackageClass)) Nil
+      else this :: denot.owner.enclClassChain
+
+    override def derivedValueClassUnbox(implicit ctx: Context): Symbol =
+    // (info.decl(nme.unbox)) orElse      uncomment once we accept unbox methods
+      (denot.info.decls.find(_.denot.is(Flags.ParamAccessor | Flags.Method)))
+
   }
 
   class ErrorSymbol(val underlying: Symbol, msg: => String)(implicit ctx: Context) extends Symbol(NoCoord, ctx.nextId) {
     type ThisName = underlying.ThisName
     denot = underlying.denot
+
+    override def enclClassChain(implicit ctx: Context): List[Symbol] = Nil
   }
 
   @sharable object NoSymbol extends Symbol(NoCoord, 0) {
     denot = NoDenotation
 
     override def associatedFile(implicit ctx: Context): AbstractFile = NoSource.file
+
+    override def enclClassChain(implicit ctx: Context): List[Symbol] = Nil
   }
 
```

Comment:
So overall, none of the additions to Symbols are needed (which is good, Symbols should be kept small). If you need functionality it's better to add it to `SymDenotations`, or if it's transform-specific even better to `SymUtils`.